pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -296,6 +296,19 @@ int sock_dtls_create(sock_dtls_t *sock, sock_udp_t *udp_sock,
     return 0;
 }
 
+int16_t sock_dtls_session_get_epoch(sock_dtls_t *sock, sock_dtls_session_t *session)
+{
+    assert(sock);
+    assert(session);
+
+    dtls_peer_t *peer = dtls_get_peer(sock->dtls_ctx, &session->dtls_session);
+    if (!peer) {
+        return -ENOTCONN;
+    }
+    dtls_security_parameters_t *security = dtls_security_params(peer);
```

Comment:
Is there a chance of `security` being NULL?

Reply:
No, `dtls_security_params()` always returns the first security parameter in the array (there are always two security_param objects per peer). And only the second one gets freed (except in `free_peer()`, but then the peer is not found anymore).

Reply:
But it would make sense to add a safety check here I guess - tinyDTLS may also change this at some point.

Reply:
> But it would make sense to add a safety check here I guess - tinyDTLS may also change this at some point.

Maybe an assert should be enough for this case. We still could optimize the check out, in case tinydtls never changes this, but also would be warned basically immediately (if this is added to the tests) when this returns a `NULL` pointer.

Reply:
Ping?

Reply:
This PR is a bit abandoned from my side at the moment to be honest. The argument of chrysn is valid. I tried to solve this via the aux API but for that the API needs to work with async sockets first. 

My intention with this function was to use it in gcoap for one requirement in the RFC - a requirement which libcoap also doesn't support as it seems. But still want to investigate why and have to look into californium/scandium as well. 

First I update my gcoap PR within the next 2 days. But PR is not forgotten, just postponed.